src/TextInput/TextInputFlat.tsx

```diff
@@ -0,0 +1,203 @@
+      textAlignVertical = 'top',
+      maxLength,
+      numberOfLines,
+      keyboardType,
+      errorText,
+      errorProps,
+      labelStyle,
+      autoCapitalize = 'none',
+      autoCorrect,
+      autoFocus,
+      multiline,
+      placeholderTextColor,
+      returnKeyType,
+      secureTextEntry,
+      onChangeText,
+      onFocus,
+      onSubmitEditing,
+      onBlur,
+    },
+    ref,
+  ) => {
+    const topAnimated = useRef<Animated.Value>(new Animated.Value(0)).current
+    const containerHeightRef = useRef<number>(metrics.huge)
+
+    const [isFocused, setIsFocused] = useState<boolean>(false)
+
+    const handleFocus = useCallback(() => {
+      setIsFocused(true)
+    }, [])
+
+    const handleBlur = useCallback(() => {
+      setIsFocused(false)
+    }, [])
+
+    const animatedFocus = useCallback(() => {
+      Animated.timing(topAnimated, {
+        toValue: 1,
+        duration: 200,
+        useNativeDriver: false,
+      }).start()
+    }, [topAnimated])
+
+    const animatedBlur = useCallback(() => {
+      Animated.timing(topAnimated, {
+        toValue: 0,
+        duration: 200,
+        useNativeDriver: false,
+      }).start()
+    }, [topAnimated])
+
+    useEffect(() => {
+      if (isFocused || value !== '') {
+        animatedFocus()
+      } else {
+        animatedBlur()
+      }
+    }, [animatedBlur, animatedFocus, isFocused, value])
+
+    const getHeight = (event: LayoutChangeEvent) => {
+      containerHeightRef.current = event.nativeEvent.layout.height
+    }
+
+    const moveToTop = useMemo(
+      () =>
+        topAnimated.interpolate({
+          inputRange: [0, 1],
+          outputRange: [(containerHeightRef.current - metrics.medium) / 2, -metrics.medium / 2],
+        }),
+      [topAnimated],
+    )
+
+    const moveToLeft = useMemo(
+      () =>
+        topAnimated.interpolate({
+          inputRange: [0, 1],
+          outputRange: [0, -metrics.medium],
+        }),
+      [topAnimated],
+    )
+
+    const reduceFontSize = useMemo(
+      () =>
+        topAnimated.interpolate({
+          inputRange: [0, 1],
+          outputRange: [responsiveFont(14), responsiveFont(11)],
+        }),
+      [topAnimated],
+    )
+
+    return (
+      <View style={containerStyle}>
+        <TextInputContainer
+          as={View}
+          onLayout={getHeight}
+          style={inputContainerStyle}
+          multiline={multiline}
+          isFocused={isFocused}>
+          {!!leftComponent && leftComponent}
+          <ContentContainer>
+            {!!label && (
+              <Animated.Text
+                style={[
+                  styles.labelContainer,
+                  {
+                    left: leftComponent ? moveToLeft : metrics.xxs,
+                    top: moveToTop,
+                    fontSize: reduceFontSize,
+                    color: isFocused ? colors.primary : colors.placeHolderText,
+                  },
+                  labelStyle,
+                ]}
```

Comment:
should use `StyleSheet.flatten` for this, if you pass custom `labelStyle` in here